setup.md

```diff
@@ -0,0 +1,92 @@
+### Install mbed-cli to a venv
```

Comment:
Why is this `setup.md` and not `README.md`?

Reply:
It's for developers. README is for users. We could rename to HACKING.md or CONTRIBUTING.md.